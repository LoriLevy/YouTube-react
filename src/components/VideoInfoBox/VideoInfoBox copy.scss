@import '../../styles/shared.scss';

.video-info-box {
    display: grid;
    grid: auto auto / 58px auto max-content; /* 2 rows / 3 cols  ( & the last never wraps) */
    align-items: top;
    grid-row-gap: 16px;
    padding-left: 15px;

    .channel-image {
        grid-row: 1 / 2;
        grid-column: 1 / 2;
    }

    .video-info {
        grid-row: 1 / 2;
        grid-column: 2 / 3;

        .subscribe {
            grid-row: 1 / 2;
            grid-column: 3 / 4;
        }
    
        .channel-name {
            font-weight: 600;
            color: #707070;
        }

        .video-pub-date {
            font-size: 13px;
            color: #707070;
        }
        
        p { 
            line-height: 1.4rem;
            margin-bottom: 0 ;
        }

        .collapsed{
            max-height: 2.8rem; /* 2 lines */
            overflow-y: hidden;
        }

        .expanded{
            max-height: none;
        }
    }

    

    .video-description {
        grid-row: 2 / 3;
        grid-column: 2 / 3;
        max-width: 600px;
    
        button {
            margin: 8px 0;
            justify-content: end;
        }
    }
}