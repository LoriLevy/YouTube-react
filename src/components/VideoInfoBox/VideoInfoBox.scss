/* File: components/VideoInfoBox/VideoInfoBox.scss */
@import "../../styles/shared.scss";

.video-info-box {
  display: grid;
  /* grid: auto auto / 58px auto max-content; /* 2 rows / 3 cols  ( & the last never wraps) */

  grid: auto auto / calc(#{$avatar-diameter} + #{$avatar-margin}) auto max-content;   /* #{$avatar-margin} convers str to num */
  align-items: top;
  grid-row-gap: 16px;
  padding-left: 15px;

  .channel-image {
    grid-row: 1 / 2;
    grid-column: 1 / 2;
  }

  .video-info {
    grid-row: 1 / 2;
    grid-column: 2 / 3;

    .subscribe {
      grid-row: 1 / 2;
      grid-column: 3 / 4;
    }

    .channel-name {
      font-weight: 600;
      color: #707070;
    }

    .video-pub-date {
      font-size: 13px;
      color: #707070;
    }

    button.button.youtube {
      float: right;
    }

    p {
      line-height: 1.4rem;
    }

    .collapsed {
      max-height: 2.8rem !important; /* 2 lines */
      overflow-y: hidden;
    }

    .expanded {
      max-height: none;
    }
  }

  .video-description {
    grid-row: 2 / 3;
    grid-column: 2 / 3;
    max-width: 600px;
    margin-bottom: 1rem;

    button {
      margin: 8px 0;
    }
  }
}
