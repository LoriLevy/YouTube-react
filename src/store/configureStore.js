import {applyMiddleware, createStore, compose} from 'redux';
import reducer from './reducers';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './sagas';

export function configureStore() {
    const sagaMiddleware = createSagaMiddleware();
    //added composeEnhancers for seeing history of actions
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const store = createStore(reducer, 
        composeEnhancers(applyMiddleware(sagaMiddleware))
    );

    sagaMiddleware.run(rootSaga);
    return store;
}

/* returns __REDUX_DEVTOOLS_EXTENSION_COMPOSE__ */