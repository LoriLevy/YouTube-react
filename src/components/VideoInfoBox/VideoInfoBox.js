import React from 'react';
import './VideoInfoBox.scss';
import {Image, Button} from 'semantic-ui-react';

export class VideoInfoBox extends React.Component {
    /* add local state: collapsed  */
    constructor(props) {
        super(props);
        this.state = {
            collapsed: true,
        };
    }
    
    
    render() {
let descriptionTextClass = 'collapsed';
let buttonTitle = 'Show More';
if(this.state.collapsed) {
    descriptionTextClass = 'expanded';
    buttonTitle = 'Show Less';
}

        return (
            <div className="video-info-box">
                <Image clasName="channel-image" src='http://via.placeholder.com/48x48' circular/>
                <div className='video-info'>
                    <div className='channel-name'>Channel Name</div>
                    <div className='video-pub-date'>Tuesday, June 25, 2019</div>
                    <Button color='youtube'>61.8k Subscribe</Button>

                    <div className='video-description'>
                        <div className={descriptionTextClass}>
                        <p>Aenean suscipit dapibus sollicitudin. Maecenas elementum imperdiet sapien at pellentesque. Quisque ornare rhoncus ex nec tristique. Integer massa ante, vehicula ut augue sed, tincidunt facilisis tellus. Sed augue tortor, dictum a ante sed, ultricies tincidunt tortor. Proin ultricies mi vel neque commodo mattis. Nullam neque ipsum, elementum in ipsum consectetur, gravida semper lectus. Curabitur at malesuada lacus, nec gravida diam. Nullam dignissim, odio sed mollis eleifend, felis turpis dictum urna, vel molestie neque nibh ut massa. Phasellus ut blandit ex. Phasellus erat leo, viverra sed commodo vitae, lacinia sit amet neque. Morbi eu dictum massa.</p>
                        <p>Duis pellentesque at quam sit amet posuere. Proin placerat, ante id ullamcorper malesuada, dui turpis bibendum ipsum, nec venenatis nisl urna vel erat. Suspendisse vestibulum risus dolor, luctus cursus tortor porta eu. Pellentesque scelerisque lectus dui, sit amet volutpat enim varius vitae. Ut ac hendrerit felis. </p>
                        <p>Vestibulum rutrum rhoncus magna sit amet tristique. Morbi nisi enim, suscipit ut varius a, congue a mauris. Sed nec lorem quis neque eleifend vehicula id vel nulla. Donec nec ipsum nec diam euismod congue quis vel turpis. Mauris eget semper quam. Curabitur vestibulum sed dolor vitae interdum.</p>
                        <p>Etiam vitae lorem non tellus sagittis ultrices sed et nisi. Phasellus blandit nec nisl in scelerisque. Ut ac purus quis neque placerat ullamcorper nec ut augue. Nam ut justo rutrum mauris lacinia tincidunt id ut purus. Integer ac efficitur est. </p>
                        <p>Proin elementum iaculis lacus, nec convallis felis dignissim et. Donec sed massa a tortor efficitur rhoncus sit amet sit amet ex. Vivamus consectetur commodo turpis sed egestas. Etiam non metus ut velit egestas congue ut sit amet erat. Maecenas id dui quis ex tempus sollicitudin in ut ante. Sed ullamcorper maximus mollis. Mauris risus nibh, dapibus ut ultricies vitae, tincidunt ac nisl.</p>
                        </div>
                    </div>

                    <Button compact onClick={this.onToggleCollapseButtonClick}>{buttonTitle}</Button>
                </div>
            </div>
        );
    }

    onToggleCollapseButtonClick = () => {
        /* forces component to rerender */
        this.setState((prevState) => {
            return{ 
                collapsed: !prevState.collapsed
            };
        });
    }
}