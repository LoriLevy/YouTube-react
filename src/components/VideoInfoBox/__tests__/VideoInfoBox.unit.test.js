import VideoInfoBox from '../VideoInfoBox';
import { shallow } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {  Provider  } from 'react-redux';

const middleware = [thunk];
const mockStore = configureStore(middleware);
const initialState = {
  example: {}
};
const store = mockStore(initialState);

describe('renders', () => {
    test('renders', () => {
        const wrapper = shallow(<Provider store={store}><VideoInfoBox /></Provider>);
        expect(wrapper).toMatchSnapshot();
    });
});


/*      // test('renders collapsed', () => {
    // const wrapper = shallow(<VideoInfoBox/>);
    // expect(wrapper).toMatchSnapshot();
    // });

    // test('renders expanded', () => {
    //     const wrapper = shallow(<VideoInfoBox/>);
        
    //     wrapper.setState({collapsed: false});
    //     expect(wrapper).toMatchSnapshot();
    //     });
    
    
    */