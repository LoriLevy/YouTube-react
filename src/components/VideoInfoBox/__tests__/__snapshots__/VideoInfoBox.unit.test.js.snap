// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="video-info-box"
/>
`;

exports[`VideoInfoBox ... renders collapsed 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    clasName="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-pub-date"
    >
      Tuesday, June 25, 2019
    </div>
    <Button
      as="button"
      color="youtube"
    >
      61.8k Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="expanded"
      >
        <p>
          Aenean suscipit dapibus sollicitudin. Maecenas elementum imperdiet sapien at pellentesque. Quisque ornare rhoncus ex nec tristique. Integer massa ante, vehicula ut augue sed, tincidunt facilisis tellus. Sed augue tortor, dictum a ante sed, ultricies tincidunt tortor. Proin ultricies mi vel neque commodo mattis. Nullam neque ipsum, elementum in ipsum consectetur, gravida semper lectus. Curabitur at malesuada lacus, nec gravida diam. Nullam dignissim, odio sed mollis eleifend, felis turpis dictum urna, vel molestie neque nibh ut massa. Phasellus ut blandit ex. Phasellus erat leo, viverra sed commodo vitae, lacinia sit amet neque. Morbi eu dictum massa.
        </p>
        <p>
          Duis pellentesque at quam sit amet posuere. Proin placerat, ante id ullamcorper malesuada, dui turpis bibendum ipsum, nec venenatis nisl urna vel erat. Suspendisse vestibulum risus dolor, luctus cursus tortor porta eu. Pellentesque scelerisque lectus dui, sit amet volutpat enim varius vitae. Ut ac hendrerit felis. 
        </p>
        <p>
          Vestibulum rutrum rhoncus magna sit amet tristique. Morbi nisi enim, suscipit ut varius a, congue a mauris. Sed nec lorem quis neque eleifend vehicula id vel nulla. Donec nec ipsum nec diam euismod congue quis vel turpis. Mauris eget semper quam. Curabitur vestibulum sed dolor vitae interdum.
        </p>
        <p>
          Etiam vitae lorem non tellus sagittis ultrices sed et nisi. Phasellus blandit nec nisl in scelerisque. Ut ac purus quis neque placerat ullamcorper nec ut augue. Nam ut justo rutrum mauris lacinia tincidunt id ut purus. Integer ac efficitur est. 
        </p>
        <p>
          Proin elementum iaculis lacus, nec convallis felis dignissim et. Donec sed massa a tortor efficitur rhoncus sit amet sit amet ex. Vivamus consectetur commodo turpis sed egestas. Etiam non metus ut velit egestas congue ut sit amet erat. Maecenas id dui quis ex tempus sollicitudin in ut ante. Sed ullamcorper maximus mollis. Mauris risus nibh, dapibus ut ultricies vitae, tincidunt ac nisl.
        </p>
      </div>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show Less
    </Button>
  </div>
</div>
`;

exports[`VideoInfoBox ... renders collapsed 2`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    clasName="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-pub-date"
    >
      Tuesday, June 25, 2019
    </div>
    <Button
      as="button"
      color="youtube"
    >
      61.8k Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="expanded"
      >
        <p>
          Aenean suscipit dapibus sollicitudin. Maecenas elementum imperdiet sapien at pellentesque. Quisque ornare rhoncus ex nec tristique. Integer massa ante, vehicula ut augue sed, tincidunt facilisis tellus. Sed augue tortor, dictum a ante sed, ultricies tincidunt tortor. Proin ultricies mi vel neque commodo mattis. Nullam neque ipsum, elementum in ipsum consectetur, gravida semper lectus. Curabitur at malesuada lacus, nec gravida diam. Nullam dignissim, odio sed mollis eleifend, felis turpis dictum urna, vel molestie neque nibh ut massa. Phasellus ut blandit ex. Phasellus erat leo, viverra sed commodo vitae, lacinia sit amet neque. Morbi eu dictum massa.
        </p>
        <p>
          Duis pellentesque at quam sit amet posuere. Proin placerat, ante id ullamcorper malesuada, dui turpis bibendum ipsum, nec venenatis nisl urna vel erat. Suspendisse vestibulum risus dolor, luctus cursus tortor porta eu. Pellentesque scelerisque lectus dui, sit amet volutpat enim varius vitae. Ut ac hendrerit felis. 
        </p>
        <p>
          Vestibulum rutrum rhoncus magna sit amet tristique. Morbi nisi enim, suscipit ut varius a, congue a mauris. Sed nec lorem quis neque eleifend vehicula id vel nulla. Donec nec ipsum nec diam euismod congue quis vel turpis. Mauris eget semper quam. Curabitur vestibulum sed dolor vitae interdum.
        </p>
        <p>
          Etiam vitae lorem non tellus sagittis ultrices sed et nisi. Phasellus blandit nec nisl in scelerisque. Ut ac purus quis neque placerat ullamcorper nec ut augue. Nam ut justo rutrum mauris lacinia tincidunt id ut purus. Integer ac efficitur est. 
        </p>
        <p>
          Proin elementum iaculis lacus, nec convallis felis dignissim et. Donec sed massa a tortor efficitur rhoncus sit amet sit amet ex. Vivamus consectetur commodo turpis sed egestas. Etiam non metus ut velit egestas congue ut sit amet erat. Maecenas id dui quis ex tempus sollicitudin in ut ante. Sed ullamcorper maximus mollis. Mauris risus nibh, dapibus ut ultricies vitae, tincidunt ac nisl.
        </p>
      </div>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show Less
    </Button>
  </div>
</div>
`;

exports[`VideoInfoBox ... renders expanded 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    clasName="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-pub-date"
    >
      Tuesday, June 25, 2019
    </div>
    <Button
      as="button"
      color="youtube"
    >
      61.8k Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="expanded"
      >
        <p>
          Aenean suscipit dapibus sollicitudin. Maecenas elementum imperdiet sapien at pellentesque. Quisque ornare rhoncus ex nec tristique. Integer massa ante, vehicula ut augue sed, tincidunt facilisis tellus. Sed augue tortor, dictum a ante sed, ultricies tincidunt tortor. Proin ultricies mi vel neque commodo mattis. Nullam neque ipsum, elementum in ipsum consectetur, gravida semper lectus. Curabitur at malesuada lacus, nec gravida diam. Nullam dignissim, odio sed mollis eleifend, felis turpis dictum urna, vel molestie neque nibh ut massa. Phasellus ut blandit ex. Phasellus erat leo, viverra sed commodo vitae, lacinia sit amet neque. Morbi eu dictum massa.
        </p>
        <p>
          Duis pellentesque at quam sit amet posuere. Proin placerat, ante id ullamcorper malesuada, dui turpis bibendum ipsum, nec venenatis nisl urna vel erat. Suspendisse vestibulum risus dolor, luctus cursus tortor porta eu. Pellentesque scelerisque lectus dui, sit amet volutpat enim varius vitae. Ut ac hendrerit felis. 
        </p>
        <p>
          Vestibulum rutrum rhoncus magna sit amet tristique. Morbi nisi enim, suscipit ut varius a, congue a mauris. Sed nec lorem quis neque eleifend vehicula id vel nulla. Donec nec ipsum nec diam euismod congue quis vel turpis. Mauris eget semper quam. Curabitur vestibulum sed dolor vitae interdum.
        </p>
        <p>
          Etiam vitae lorem non tellus sagittis ultrices sed et nisi. Phasellus blandit nec nisl in scelerisque. Ut ac purus quis neque placerat ullamcorper nec ut augue. Nam ut justo rutrum mauris lacinia tincidunt id ut purus. Integer ac efficitur est. 
        </p>
        <p>
          Proin elementum iaculis lacus, nec convallis felis dignissim et. Donec sed massa a tortor efficitur rhoncus sit amet sit amet ex. Vivamus consectetur commodo turpis sed egestas. Etiam non metus ut velit egestas congue ut sit amet erat. Maecenas id dui quis ex tempus sollicitudin in ut ante. Sed ullamcorper maximus mollis. Mauris risus nibh, dapibus ut ultricies vitae, tincidunt ac nisl.
        </p>
      </div>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show Less
    </Button>
  </div>
</div>
`;

exports[`VideoInfoBox ...renders collapsed 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    clasName="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-pub-date"
    >
      Tuesday, June 25, 2019
    </div>
    <Button
      as="button"
      color="youtube"
    >
      61.8k Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="expanded"
      >
        <p>
          Aenean suscipit dapibus sollicitudin. Maecenas elementum imperdiet sapien at pellentesque. Quisque ornare rhoncus ex nec tristique. Integer massa ante, vehicula ut augue sed, tincidunt facilisis tellus. Sed augue tortor, dictum a ante sed, ultricies tincidunt tortor. Proin ultricies mi vel neque commodo mattis. Nullam neque ipsum, elementum in ipsum consectetur, gravida semper lectus. Curabitur at malesuada lacus, nec gravida diam. Nullam dignissim, odio sed mollis eleifend, felis turpis dictum urna, vel molestie neque nibh ut massa. Phasellus ut blandit ex. Phasellus erat leo, viverra sed commodo vitae, lacinia sit amet neque. Morbi eu dictum massa.
        </p>
        <p>
          Duis pellentesque at quam sit amet posuere. Proin placerat, ante id ullamcorper malesuada, dui turpis bibendum ipsum, nec venenatis nisl urna vel erat. Suspendisse vestibulum risus dolor, luctus cursus tortor porta eu. Pellentesque scelerisque lectus dui, sit amet volutpat enim varius vitae. Ut ac hendrerit felis. 
        </p>
        <p>
          Vestibulum rutrum rhoncus magna sit amet tristique. Morbi nisi enim, suscipit ut varius a, congue a mauris. Sed nec lorem quis neque eleifend vehicula id vel nulla. Donec nec ipsum nec diam euismod congue quis vel turpis. Mauris eget semper quam. Curabitur vestibulum sed dolor vitae interdum.
        </p>
        <p>
          Etiam vitae lorem non tellus sagittis ultrices sed et nisi. Phasellus blandit nec nisl in scelerisque. Ut ac purus quis neque placerat ullamcorper nec ut augue. Nam ut justo rutrum mauris lacinia tincidunt id ut purus. Integer ac efficitur est. 
        </p>
        <p>
          Proin elementum iaculis lacus, nec convallis felis dignissim et. Donec sed massa a tortor efficitur rhoncus sit amet sit amet ex. Vivamus consectetur commodo turpis sed egestas. Etiam non metus ut velit egestas congue ut sit amet erat. Maecenas id dui quis ex tempus sollicitudin in ut ante. Sed ullamcorper maximus mollis. Mauris risus nibh, dapibus ut ultricies vitae, tincidunt ac nisl.
        </p>
      </div>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show Less
    </Button>
  </div>
</div>
`;

exports[`VideoInfoBox it renders 1`] = `
<div
  className="video-info-box"
/>
`;

exports[`VideoInfoBox it renders collapsed 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    clasName="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-pub-date"
    >
      Tuesday, June 25, 2019
    </div>
    <Button
      as="button"
      color="youtube"
    >
      61.8k Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="expanded"
      >
        <p>
          Aenean suscipit dapibus sollicitudin. Maecenas elementum imperdiet sapien at pellentesque. Quisque ornare rhoncus ex nec tristique. Integer massa ante, vehicula ut augue sed, tincidunt facilisis tellus. Sed augue tortor, dictum a ante sed, ultricies tincidunt tortor. Proin ultricies mi vel neque commodo mattis. Nullam neque ipsum, elementum in ipsum consectetur, gravida semper lectus. Curabitur at malesuada lacus, nec gravida diam. Nullam dignissim, odio sed mollis eleifend, felis turpis dictum urna, vel molestie neque nibh ut massa. Phasellus ut blandit ex. Phasellus erat leo, viverra sed commodo vitae, lacinia sit amet neque. Morbi eu dictum massa.
        </p>
        <p>
          Duis pellentesque at quam sit amet posuere. Proin placerat, ante id ullamcorper malesuada, dui turpis bibendum ipsum, nec venenatis nisl urna vel erat. Suspendisse vestibulum risus dolor, luctus cursus tortor porta eu. Pellentesque scelerisque lectus dui, sit amet volutpat enim varius vitae. Ut ac hendrerit felis. 
        </p>
        <p>
          Vestibulum rutrum rhoncus magna sit amet tristique. Morbi nisi enim, suscipit ut varius a, congue a mauris. Sed nec lorem quis neque eleifend vehicula id vel nulla. Donec nec ipsum nec diam euismod congue quis vel turpis. Mauris eget semper quam. Curabitur vestibulum sed dolor vitae interdum.
        </p>
        <p>
          Etiam vitae lorem non tellus sagittis ultrices sed et nisi. Phasellus blandit nec nisl in scelerisque. Ut ac purus quis neque placerat ullamcorper nec ut augue. Nam ut justo rutrum mauris lacinia tincidunt id ut purus. Integer ac efficitur est. 
        </p>
        <p>
          Proin elementum iaculis lacus, nec convallis felis dignissim et. Donec sed massa a tortor efficitur rhoncus sit amet sit amet ex. Vivamus consectetur commodo turpis sed egestas. Etiam non metus ut velit egestas congue ut sit amet erat. Maecenas id dui quis ex tempus sollicitudin in ut ante. Sed ullamcorper maximus mollis. Mauris risus nibh, dapibus ut ultricies vitae, tincidunt ac nisl.
        </p>
      </div>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show Less
    </Button>
  </div>
</div>
`;

exports[`VideoInfoBox renders collapsed 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    clasName="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-pub-date"
    >
      Tuesday, June 25, 2019
    </div>
    <Button
      as="button"
      color="youtube"
    >
      61.8k Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="expanded"
      >
        <p>
          Aenean suscipit dapibus sollicitudin. Maecenas elementum imperdiet sapien at pellentesque. Quisque ornare rhoncus ex nec tristique. Integer massa ante, vehicula ut augue sed, tincidunt facilisis tellus. Sed augue tortor, dictum a ante sed, ultricies tincidunt tortor. Proin ultricies mi vel neque commodo mattis. Nullam neque ipsum, elementum in ipsum consectetur, gravida semper lectus. Curabitur at malesuada lacus, nec gravida diam. Nullam dignissim, odio sed mollis eleifend, felis turpis dictum urna, vel molestie neque nibh ut massa. Phasellus ut blandit ex. Phasellus erat leo, viverra sed commodo vitae, lacinia sit amet neque. Morbi eu dictum massa.
        </p>
        <p>
          Duis pellentesque at quam sit amet posuere. Proin placerat, ante id ullamcorper malesuada, dui turpis bibendum ipsum, nec venenatis nisl urna vel erat. Suspendisse vestibulum risus dolor, luctus cursus tortor porta eu. Pellentesque scelerisque lectus dui, sit amet volutpat enim varius vitae. Ut ac hendrerit felis. 
        </p>
        <p>
          Vestibulum rutrum rhoncus magna sit amet tristique. Morbi nisi enim, suscipit ut varius a, congue a mauris. Sed nec lorem quis neque eleifend vehicula id vel nulla. Donec nec ipsum nec diam euismod congue quis vel turpis. Mauris eget semper quam. Curabitur vestibulum sed dolor vitae interdum.
        </p>
        <p>
          Etiam vitae lorem non tellus sagittis ultrices sed et nisi. Phasellus blandit nec nisl in scelerisque. Ut ac purus quis neque placerat ullamcorper nec ut augue. Nam ut justo rutrum mauris lacinia tincidunt id ut purus. Integer ac efficitur est. 
        </p>
        <p>
          Proin elementum iaculis lacus, nec convallis felis dignissim et. Donec sed massa a tortor efficitur rhoncus sit amet sit amet ex. Vivamus consectetur commodo turpis sed egestas. Etiam non metus ut velit egestas congue ut sit amet erat. Maecenas id dui quis ex tempus sollicitudin in ut ante. Sed ullamcorper maximus mollis. Mauris risus nibh, dapibus ut ultricies vitae, tincidunt ac nisl.
        </p>
      </div>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show Less
    </Button>
  </div>
</div>
`;

exports[`VideoInfoBox renders expanded 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    clasName="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Channel Name
    </div>
    <div
      className="video-pub-date"
    >
      Tuesday, June 25, 2019
    </div>
    <Button
      as="button"
      color="youtube"
    >
      61.8k Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="collapsed"
      >
        <p>
          Aenean suscipit dapibus sollicitudin. Maecenas elementum imperdiet sapien at pellentesque. Quisque ornare rhoncus ex nec tristique. Integer massa ante, vehicula ut augue sed, tincidunt facilisis tellus. Sed augue tortor, dictum a ante sed, ultricies tincidunt tortor. Proin ultricies mi vel neque commodo mattis. Nullam neque ipsum, elementum in ipsum consectetur, gravida semper lectus. Curabitur at malesuada lacus, nec gravida diam. Nullam dignissim, odio sed mollis eleifend, felis turpis dictum urna, vel molestie neque nibh ut massa. Phasellus ut blandit ex. Phasellus erat leo, viverra sed commodo vitae, lacinia sit amet neque. Morbi eu dictum massa.
        </p>
        <p>
          Duis pellentesque at quam sit amet posuere. Proin placerat, ante id ullamcorper malesuada, dui turpis bibendum ipsum, nec venenatis nisl urna vel erat. Suspendisse vestibulum risus dolor, luctus cursus tortor porta eu. Pellentesque scelerisque lectus dui, sit amet volutpat enim varius vitae. Ut ac hendrerit felis. 
        </p>
        <p>
          Vestibulum rutrum rhoncus magna sit amet tristique. Morbi nisi enim, suscipit ut varius a, congue a mauris. Sed nec lorem quis neque eleifend vehicula id vel nulla. Donec nec ipsum nec diam euismod congue quis vel turpis. Mauris eget semper quam. Curabitur vestibulum sed dolor vitae interdum.
        </p>
        <p>
          Etiam vitae lorem non tellus sagittis ultrices sed et nisi. Phasellus blandit nec nisl in scelerisque. Ut ac purus quis neque placerat ullamcorper nec ut augue. Nam ut justo rutrum mauris lacinia tincidunt id ut purus. Integer ac efficitur est. 
        </p>
        <p>
          Proin elementum iaculis lacus, nec convallis felis dignissim et. Donec sed massa a tortor efficitur rhoncus sit amet sit amet ex. Vivamus consectetur commodo turpis sed egestas. Etiam non metus ut velit egestas congue ut sit amet erat. Maecenas id dui quis ex tempus sollicitudin in ut ante. Sed ullamcorper maximus mollis. Mauris risus nibh, dapibus ut ultricies vitae, tincidunt ac nisl.
        </p>
      </div>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show More
    </Button>
  </div>
</div>
`;
